src/main.rs

```diff
@@ -21,9 +21,6 @@ const OTHER_BACKEND_NAME: &str = "other_backend_name";
 /// If `main` returns an error, a 500 error response will be delivered to the client.
 #[fastly::main]
 fn main(mut req: Request) -> Result<Response, Error> {
-    // Make any desired changes to the client request.
-    req.set_header(header::HOST, "example.com");
```

Comment:
Due to the change in 'default' backends I needed to move the change of Host header down the file as the first backend (developer.fastly.com) will fail if we change the host. 

I could have just deleted the change of Host but I thought it worth keeping it for the sake of the 'starter kit' aspect (in case the user needed to know how to do that, then having it in this example code would be useful, especially as I can imagine it being a common thing for users to want to do).

Reply:
The host header will reflect wherever the application is running, e.g. `woefully-obtuse-aristocrat.edgecompute.app` — I'm not sure that will work?

```
$ curl -I -H 'Host: woefully-obtuse-aristocrat.edgecompute.app' https://developer.fastly.com/solutions/
HTTP/2 500
server: Varnish
retry-after: 0
content-type: text/html
cache-control: private, no-cache
x-served-by: cache-hhn4038-HHN
accept-ranges: bytes
date: Wed, 28 Jul 2021 09:42:56 GMT
via: 1.1 varnish
content-length: 273
```

Reply:
Yup, this is why I moved it down to the 'other backend' code condition block as the _default_ expectation for that backend would be httpbin.org which does accept a Host like example.com

Reply:
Just noting here that I think we settled on having the CLI always configure a host_header override when creating the backend so we no longer recommend that you set backend-specific host headers in edge code.